ASP/HTML-ASP.sublime-syntax

```diff
@@ -14,9 +14,10 @@ contexts:
         - match: "%>"
           scope: punctuation.section.embedded.end.asp
           pop: true
-        - match: (').*?(?=%>)
-          scope: comment.line.apostrophe.asp
-          captures:
-            1: punctuation.definition.comment.asp
-        - include: scope:source.asp
+        - match: ''
+          push:
+            - include: scope:source.asp # don't directly push, so that the scope doesn't also contain 'source.asp ' as well as our 'source.asp.embedded.html '
+          with_prototype:
+            - match: "(?=%>)"
+              pop: true
```

Comment:
I am invested in how this goes, because a C# HTML syntax I'm working on has a similar problem. (I've taken some cues from the ASP HTML syntax and some from the PHP syntax.) This is a valid pattern in that language:

```html
<div>
<% if (condition) { %>
    <p>Hello World</p>
<% } %>
</div>
```

The problem is that C# doesn't give the scope back to HTML in the `if{...}` body, even though the C# scope should be closed with `%>`. If you guys take a route with ASP that makes such patterns possible, I'm going to model the C# HTML after it.

I can't quite tell what @wbond is saying. 😕